File: JavaScript/Big-O/space-complexity.js
// What causes Space complexity:

// Variables
// Data Structures
// Function Call
// Allocations

function booo(n) {
  for (let i = 0; i < n.length; i++) {
    console.log('boooo!');
  }
}

booo([1, 2, 3, 4, 5]); // O(1)

function arrayOfHiNTimes(n) {
  let hiArray = [];
  for (let i = 0; i < n; i++) {
    hiArray[i] = 'hi';  // allocates memory
  }
  console.log(hiArray)
  return hiArray;
}

arrayOfHiNTimes(6); // O(n)

File: Python/Big-O/space_complexity.py
# What causes Space complexity:

# Variables
# Data Structures
# Function Call
# Allocations


def booo(n):
    for _ in range(len(n)):
        print('boooo!')


booo([1, 2, 3, 4, 5])  # O(1)


def arrayOfHiNTimes(n):
    hiArray = []
    for i in range(len(n)):
        hiArray[i] = 'hi'  # allocates memory
    print(hiArray)
    return hiArray


arrayOfHiNTimes(6)  # O(n)
